pkg/daemon/manager/manager.go

```diff
@@ -37,6 +39,14 @@ func (m *manager) OpenService(name string) (winsvc.Service, error) {
 	underlyingMgr := (*mgr.Mgr)(m)
 	return underlyingMgr.OpenService(name)
 }
+func (m *manager) DeleteService(name string) error {
+	underlyingMgr := (*mgr.Mgr)(m)
+	winSvc, err := underlyingMgr.OpenService(name)
+	if err != nil {
+		return err
+	}
+	return winSvc.Delete()
+}
```

Comment:
Please write a test case for manually testing this to ensure:
1) This doesn't error out if another service handle is floating around due to `DeleteService(service)` being called within the same function as `OpenService(service)` 
2) The existence of other open handles for that service does not cause the service to be stuck in waiting for deletion.

Reply:
Sure, I added this within the `manager.DeleteService()` [unit tests](https://github.com/openshift/windows-machine-config-operator/pull/1166/commits/8dc05cb9404101921b14e57ba677b21408db22e5#diff-770772d9b994c813411fdf9f00b851ef0e3a7b13e05398d09fce6a9d120a95b4R165-R171)

Reply:
I meant this needs to be tested with the Windows service manager, not with our fake manager. I don't know the behavior of the real manager in this scenario.

Reply:
I can confirm that there's no issue opening a service multiple times in a tow when using the real manager. Ran on Windows machine without error and printed expected msg:
```golang
func runTest() {
	svcMgr, err := manager.New()
	if err != nil {
		klog.Exitf("could not create service manager: %s", err.Error())
	}

	_, err = svcMgr.CreateService("testsvc", "testsvc", mgr.Config{
		Description: "testsvc",
	})
	if err != nil {
		klog.Exitf("could not create: %s", err.Error())
	}
	_, err = svcMgr.OpenService("testsvc")
	if err != nil {
		klog.Exitf("could not open: %s", err.Error())
	}
	klog.Info("open1")
        _, err = svcMgr.OpenService("testsvc")
	if err != nil {
		klog.Exitf("could not open: %s", err.Error())
	}
	klog.Info("open2")

	err = svcMgr.DeleteService("testsvc") // <--------- has additional underlying OpenService call
	if err != nil {
		klog.Exitf("could not delete: %s", err.Error())
	}
	klog.Info("end")
}
```
Additionally the `testsvc` doesn't appear in the output of `Get-Service` so successfully deleted

Reply:
Thanks for doing that